api.py

```diff
@@ -1,11 +1,97 @@
-from flask import Flask
+import json
+import os
+import urllib
+from json import JSONDecodeError
 
-from app.views.status import status_blueprint
-from app.views.validate import validate_blueprint
+import requests
+import uvicorn
+from fastapi import Body, FastAPI, Response
+from requests import RequestException
+from structlog import get_logger
+
+from app.validators.questionnaire_validator import QuestionnaireValidator
+
+app = FastAPI()
+
+
+@app.get("/status")
+async def status():
+    return Response(status_code=200)
+
+
+logger = get_logger()
+
+AJV_HOST = os.getenv("AJV_HOST", "localhost")
+
+AJV_VALIDATOR_URL = f"http://{AJV_HOST}:5002/validate"
+
+
+@app.post("/validate")
+async def validate_schema_request_body(payload=Body(None)):
+    logger.info("Validating schema")
+    return await validate_schema(payload)
+
+
+@app.get("/validate")
+async def validate_schema_from_url(url: str = ""):
+    if url:
+        logger.info("Validating schema from URL", url=url)
+        try:
+            with urllib.request.urlopen(url) as opened_url:
+                return await validate_schema(data=opened_url.read().decode())
+        except urllib.error.URLError:
+            return (
+                Response(
+                    status_code=404,
+                    content=f"Could not load schema at URL [{url}]",
+                ),
+            )
+
+
+async def validate_schema(data):
+    json_to_validate = None
+    if data:
+        if isinstance(data, str):
+            try:
+                json_to_validate = json.loads(data)
```

Comment:
Minor suggestion - Could do this with a ternary to avoid the `elif` at the end? Something like 
```
try:
    json_to_validate = json.loads(data) if isinstance(data, str) else data
    ...

```

Reply:
Not sure if this will work as the `else data` replaces a line where it's checked if it's a dict in the original change.

Reply:
I'm not 100% sure but if we're only parsing a str or dict then it might be okay? Looks like it was intentionally checking for dict explicitly before though like you said, so happy to leave it as it was 👍